utils/weather_tools.py

```diff
@@ -37,114 +23,230 @@ def generate_stationID_dict():
-    df_stations_US.WBAN = df_stations_US.WBAN.astype('int64')
-    df_stations_US.dropna(inplace=True)
-    df_stations_US = df_stations_US[df_stations_US.WBAN != 99999].copy()
+    cols_to_use = ['ST', 'StID']
+    df_stations = pd.read_csv(f"{DATA_DIR}/stations_TZ.csv", usecols=cols_to_use)
 
-    def my_func(row):
-        return str(row.USAF) + '-' + str(row.WBAN).zfill(5)
-    df_stations_US['stationID'] = df_stations_US.apply(my_func, axis=1)
+    states = pd.unique(df_stations.ST.values)
 
-    states = pd.unique(df_stations_US.ST.values)
     stations_in_state = {}
     for i in range(len(states)):
-        stations_in_state[states[i]] = list(
-            df_stations_US[df_stations_US.ST == states[i]].stationID.values)
+        stations_in_state[states[i]] = list(df_stations[df_stations.ST == states[i]].StID.values)
     return stations_in_state
-stations_in_state = generate_stationID_dict()
 
+stations_in_state = generate_stationID_dict()
 
-def df_from_file(stationID, year):
-    """Takes stationID and year and outputs a dataframe with Temperature
-
-    Note that some tempertaure readings are missing in general.
-    """
-
-    url_common = "ftp://ftp.ncdc.noaa.gov/pub/data/noaa/"
-    url_ftp = url_common + str(year) + "/" + \
-        stationID + "-" + str(year) + ".gz"
-
-    WBAN = re.findall(r'(.*)-(.*)', stationID)
-    WBAN = WBAN[0][1]
-
-    #df = df_tmp[0].str.extractall(r'24 HR AVG TEMP \(F\):(\s)?(?P<Temperature>\d\d)')
-
-    def extract_Temp(row):
-        result = re.findall(
-            r'24 HR AVG TEMP \(F\):(\s)?(?P<Temperature>\d\d)', row)
-        if len(result) == 0:
-            result = [(np.nan, np.nan)]
-        return result[-1][1]
-
-    def extract_date(row):
-        date = [np.nan, np.nan, np.nan]
-        date = re.findall(WBAN + r'(\d\d\d\d)(\d\d)(\d\d)', row)
-        if date != []:
-            date = date[0][0] + date[0][1] + date[0][2]
-        return date
-    try:
-        df = pd.read_csv(url_ftp, compression='gzip', header=None)
-        df['TEMPERATURE' + stationID] = df[0].apply(extract_Temp)
-        df['Date'] = df[0].apply(extract_date)
-        df.dropna(inplace=True)
-        df.drop(columns=[0], inplace=True)
-        df['Date'] = pd.to_datetime(df['Date'])
-        df['TEMPERATURE' + stationID] = df['TEMPERATURE' +
-                                           stationID].astype('int64')
-        df.reset_index(inplace=True)
-        df.drop(columns=['index'], inplace=True)
-        return df
-    except (pd.errors.ParserError, OSError, ValueError) as e:  # URLError:
-        #print('No good')
-        print(f'{e} with file: ' + url_ftp)
-        Date = pd.date_range(str(year) + '-01-01', str(year) + '-12-31')
-        df = pd.DataFrame(data=Date, columns=['Date'])
-        return df
-
+key_list = df_stations_US.StID.values
+valu_list = df_stations_US.TZone.values 
+TZ_dict = {key_list[i] : valu_list[i] for i in range(len(key_list))} # dictionary containing {Station_ID: Time-Zone}
```

Comment:
Is there a way to have these not be global variables? i.e. can they be defined in the functions that use them, or is that too slow?

If we're going to declare global variables, let's do it in an easy-to-spot place that's well-commented